experiments/PlaywrightSharp/PlaywrightSharp/ApiSurface/IBrowserContext.cs

```diff
@@ -0,0 +1,411 @@
+	
+		/// <summary>
+		/// <para>
+		/// The method adds a function called <paramref name="name"/> on the <c>window</c> object
+		/// of every frame in every page in the context. When called, the function executes
+		/// <paramref name="callback"/> and returns a [Promise] which resolves to the return
+		/// value of <paramref name="callback"/>.
+		/// </para>
+		/// <para>If the <paramref name="callback"/> returns a [Promise], it will be awaited.</para>
+		/// <para>See <see cref="IPage.ExposeFunctionAsync"/> for page-only version.</para>
+		/// <para>An example of adding an <c>md5</c> function to all pages in the context:</para>
+		/// </summary>
+		/// <param name="name">Name of the function on the window object.</param>
+		/// <param name="callback">Callback function that will be called in the Playwright's context.</param>
+		Task ExposeFunctionAsync(string name, Action callback);
+	
+		/// <summary>
+		/// <para>
+		/// Grants specified permissions to the browser context. Only grants corresponding permissions
+		/// to the given origin if specified.
+		/// </para>
+		/// </summary>
+		/// <param name="permissions">
+		/// A permission or an array of permissions to grant. Permissions can be one of the
+		/// following values:
+		/// <list type="bullet">
+		/// <item><description>`'geolocation'`</description></item>
+		/// <item><description>`'midi'`</description></item>
+		/// <item><description>`'midi-sysex'` (system-exclusive midi)</description></item>
+		/// <item><description>`'notifications'`</description></item>
+		/// <item><description>`'push'`</description></item>
+		/// <item><description>`'camera'`</description></item>
+		/// <item><description>`'microphone'`</description></item>
+		/// <item><description>`'background-sync'`</description></item>
+		/// <item><description>`'ambient-light-sensor'`</description></item>
+		/// <item><description>`'accelerometer'`</description></item>
+		/// <item><description>`'gyroscope'`</description></item>
+		/// <item><description>`'magnetometer'`</description></item>
+		/// <item><description>`'accessibility-events'`</description></item>
+		/// <item><description>`'clipboard-read'`</description></item>
+		/// <item><description>`'clipboard-write'`</description></item>
+		/// <item><description>`'payment-handler'`</description></item>
+		/// </list>
+		/// </param>
+		/// <param name="origin">The [origin] to grant permissions to, e.g. "https://example.com".</param>
+		Task GrantPermissionsAsync(string[] permissions, string origin);
+	
+		/// <summary><para>Creates a new page in the browser context.</para></summary>
+		Task<IPage> GetNewPageAsync();
+	
+		/// <summary>
+		/// <para>
+		/// Returns all open pages in the context. Non visible pages, such as <c>"background_page"</c>
+		/// will not be listed here. You can find them using <see cref="IChromiumBrowserContext.BackgroundPages"/>.
+		/// </para>
+		/// </summary>
+		dynamic GetPages();
```

Comment:
I think we should never get to the point to return a dynamic. I would make it fail.